src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -595,12 +591,30 @@ int wm_vuldet_add_cve_node(cve_vuln_pkg *newPkg, const char *cve, OSHash *cve_ta
         do {
             if (!wm_vuldet_compare_pkg(tmpPkg, newPkg)) { // Same package?
```

Comment:
If we are in front of the same package, the pointer shouldn't be directly assigned because when the method returns, the `newPkg` structure is destroyed.